### DISCUSSION TITLE
Disable Admin on Specific Server Instances via Config

### DISCUSSION THREAD

**Discussion by JonGoldmanPlaybill** [craftcms/cms#12628] [Ideas] [answered]
Would love to see a way to disable the admin in its entirety on specific servers (.env setting to general.php).

We separate our instances of public web-heads and a separate set of admin servers.  We use an AWS load balancer to use OIDC authentication for our admin servers, and then randomize the cpURL on the web-heads:
`'cpTrigger' => $randomizeCraftCpTrigger ? \craft\helpers\StringHelper::randomString(64) : 'admin'`

This prevents the traffic (and resulting database traffic) associated with brute-force password attempts by not even allowing non-SSO users access to the login page.  However even with a 64 character string as the cpURL, there is still the possibility of scanning attempts to find it and resulting brute force attempts.

A simple ability to set `'cpTrigger' => false` would be all that would be needed for this.

**Comment by brandonkelly** [comment, marked as answer]
You can define a [`CRAFT_CP`](https://craftcms.com/docs/4.x/config/#craft-cp) PHP constant or environment variable, set to `false`, to explicitly tell Craft it’s **not** an control panel request.

```php
// web/index.php

define('CRAFT_CP', false);
```

```sh
# .env

CRAFT_CP=false
```
